### DISCUSSION TITLE
Transaction#select() replacement since Jedis version 4

### DISCUSSION THREAD

**Discussion by YAMAGON4210** [xetorthio/jedis#3638] [Q&A] [answered]
Hello.

We are in the process of upgrading the Jedis dependency for one of our services from version 3.7.0 to the latest version 5.0.0.

In the process, we encountered an event that prevents us from using the previous code.

> ```// watch for key in database 0
>             jedis.watch(key);
>             jedis.select(1);
>             // watch for key2 in database 1
>             jedis.watch(key2);
>             jedis.select(0);
> 
>             Transaction transaction = jedis.multi();
> 
>             transaction.hset(key, "name", "john Doe");
>             transaction.hset(key, "class", "Worker");
> 
>             // method cannot be used.
>             transaction.select(1);
> 
>             transaction.hset(key2, "name", "jane Doe");
>             transaction.hset(key2, "class", "Worker");
>             transaction.exec();
> ```

As you can see, we used method call, but since version 4 - it is removed. What would be the closest equivalent (or is it needed at all) of the call since Jedis version 4?`transaction.select(int index)`

I would also like to know why this method was removed.

I have looked for a relevant statement in the issues and documentation, but cannot find a clear reason.
  

> Is it relevant?(https://github.com/redis/jedis/issues/2820)

Is it because it is an anti-pattern against Redis design philosophy?

**Comment by YAMAGON4210** [comment]
I consider the following method as an alternative.
`transaction.sendCommand(SELECT, toByteArray(9));`
However, I could not determine if using this method is the best answer.
I think there is some glitch or inconvenient reason why that method was removed.

I would like to minimize the impact of code changes in the recommended manner.
I would also like to know why it was removed.`transaction.select(int index)`

**Comment by uglide** [comment]
Hi @YAMAGON4210 
We do not recommend using the SELECT command, especially in transactions. That's why the API was removed in the recent versions of Jedis. As you mentioned above, you can use raw commands API to temporarily workaround it in your code. Still, I recommend you update the application logic to avoid using the SELECT command.

**Reply by YAMAGON4210** [reply]
@uglide 
Thanks for the comment.
If it is not too much trouble, I would like to know why I should not use it. 
I believe Redis is NoSQL and should not be operated or designed like an RDBMS.

**Reply by uglide** [reply, marked as answer]
The reason is quite simple. Redis Cluster doesn't support SELECT commands. Using SELECT prevents your application code from being scaled when needed with Redis Cluster.

**Reply by YAMAGON4210** [reply]
I really appreciate you answering my questions. 
Sorry if I am wrong.
I understood as follows,clustering is a Redis setting and Redis client libraries should not implement methods that depend on that setting.
